delivery_carrier_account/models/delivery_carrier.py

```diff
@@ -0,0 +1,24 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+
+import json
+
+from odoo import api, fields, models
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    carrier_account_id = fields.Many2one(
+        "carrier.account", string="Account", company_dependent=True
```

Comment:
The domain can be put statically and don't require additional modules:

```suggestion
        "carrier.account",
        string="Account",
        company_dependent=True,
        domain="[("delivery_type", "=", delivery_type)]",
```

Reply:
Thanks, I fixed it